cli/config.go

```diff
@@ -202,6 +203,21 @@ func (h hypervisor) defaultVCPUs() uint32 {
 	return uint32(h.DefaultVCPUs)
 }
 
+func (h hypervisor) defaultMaxVCPUs() uint32 {
+	numcpus := goruntime.NumCPU()
+	maxvcpus := vc.MaxQemuVCPUs()
+
+	if h.DefaultMaxVCPUs <= 0 || h.DefaultMaxVCPUs > int32(numcpus) {
+		// Don't exceed the maximum number of vCPUs supported by QEMU
+		if numcpus > int(maxvcpus) {
+			return maxvcpus
+		}
+		return uint32(numcpus)
+	}
+
+	return uint32(h.DefaultMaxVCPUs)
```

Comment:
This logic is still a bit odd to me.  What happens if default-provided is not greater than numcpus, but it is still greater than  is greater than  maxvcpus?  
e.g., max-from-hypervisor is 100, number of physical cores is 120, but our configured default is 110?

Also I think you should s/QEMU/hypervisor?

Reply:
> e.g., max-from-hypervisor is 100, number of physical cores is 120, but our configured default is 110?

to avoid issues with KVM, the maximum number of CPUs supported by KVM is used, in this example `100`

> Also I think you should s/QEMU/hypervisor?

sure, I can do it, but in another PR, wdyt?